### ISSUE TITLE
ResolveMany not work with generics after any Unregister

### ISSUE THREAD

**Issue by grifsun** [dadhi/DryIoc#412]
This code throw exception
`ˋˋ
public interface IHandler<TEvent> { } 
public class A { }
public class Service { }

var container = new Container();
container.Register<Service>();
container.Unregister<Service>();
var genericHandlers = container.ResolveMany<IHandler<A>>();
`ˋˋ

System.NullReferenceException : Object reference not set to an instance of an object.
   at DryIoc.Container.Registry.GetServiceRegistrations()+MoveNext() in C:\Develop\dryioc\src\DryIoc\Container.cs:line 2284
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at System.Collections.Generic.LargeArrayBuilder`1.AddRange(IEnumerable`1 items)
   at System.Collections.Generic.SparseArrayBuilder`1.ReserveOrAdd(IEnumerable`1 items)
   at System.Linq.Enumerable.ConcatNIterator`1.LazyToArray()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.OrderedEnumerable`1.GetEnumerator()+MoveNext()
   at DryIoc.Container.DryIoc.IResolver.ResolveMany(Type serviceType, Object serviceKey, Type requiredServiceType, Request preResolveParent, Object[] args)+MoveNext() in C:\Develop\dryioc\src\DryIoc\Container.cs:line 674
   at System.Linq.Enumerable.CastIterator[TResult](IEnumerable source)+MoveNext()
   at System.Linq.Enumerable.Count[TSource](IEnumerable`1 source)
   at DryIoc.UnitTests.OpenGenericsResolveManyTests.NotWorkExample() in C:\Develop\dryioc\test\DryIoc.UnitTests\OpenGenericsResolveManyTests.cs:line 16

**Comment by dadhi** [comment]
@grifsun Hello, I am no seeing any generics in your example and what are registering/unregistering:

```cs
public interface IHandler { }
public class A { }
public class Service { } // where are generics?

var container = new Container();
container.Register(); // ?
container.Unregister(); // ?
var genericHandlers = container.ResolveMany<IHandler>();
```

**Comment by grifsun** [comment]
Sorry, wrong formatting
That right code:

```
public interface IHandler<T> { } 
public class A { }
public class Service { }

var container = new Container();
container.Register<Service>();
container.Unregister<Service>();
var genericHandlers = container.ResolveMany<IHandler<A>>();
```

**Comment by dadhi** [comment]
@grifsun fix is out with DryIoc v4.8.1
